sds_data_manager/lambda_code/IAlirtCode/ialirt_catalog_api.py

```diff
@@ -7,25 +13,83 @@
 logger.setLevel(logging.INFO)
 
 
-def lambda_handler(event, context):
+def lambda_handler(event, context) -> dict:
     """Entry point to the catalog API lambda.
 
-    Blank function for initial setup.
+    Takes in a date range and ground station, and queries the s3 bucket that holds
+    the pointing schedules for ground stations part of the I-ALiRT network.
+
+    If files exist for the requested date range, the response body returns all file
+    versions for every date, including the file name, a pre-signed url for its download,
+    and its last date modified.
 
     Parameters
     ----------
     event : dict
         The JSON formatted document with the data required for the
-        lambda function to process
+        lambda function to process, i.e., date range and ground station
     context : LambdaContext
         This object provides methods and properties that provide
         information about the invocation, function,
         and runtime environment.
+
+    Returns
+    -------
+    dict
+        A dictionary containing headers, status code, and body, designed to be returned
+        as an API response.
     """
-    logger.info(f"Event: {event}")
-    logger.info(f"Context: {context}")
+    logger.info("Received event: " + json.dumps(event, indent=2))
+
+    if not event['start_date'] < event['end_date']:
+        response_body = ("The start date of the requested time frame is not before the "
+                         "end date. Please supply a valid time range.")
+        return {
+            "statusCode": 400,
+            "headers": {"Content-Type": "application/json"},
+            "body": json.dumps(response_body)
+        }
+
+    bucket = os.getenv("S3_BUCKET")
+    region = os.getenv("REGION")
+
+    s3_client = boto3.client(
+        "s3",
+        region_name=region,
+        config=botocore.client.Config(signature_version="s3v4"),
+    )
+
+    response_body = {}
+    for day in np.arange(event['start_date'], event['end_date'], dtype='datetime64[D]'):
+        day_path = (f"pointing_schedules/{event['station']}/" +
+                    day.item().strftime('%Y%m%d') + "/")
+        files = s3_client.list_objects_v2(Bucket=bucket, Prefix=day_path)
```

Comment:
An alternative idea to this loop over listing objects per day would be to get all of your objects in one `list_objects_v2()` call and then filter the list of objects returned after that.

Reply:
To do that, I would have to pass in a list of prefixes (dates), which would require doing the same iterating that the loop is already doing, so I'd prefer to keep the call to `list_objects_v2` within the loop logic and call it for every day. And every day will have multiple versions of the scheduling file that should be returned.

Reply:
I see what you're saying. At some point, there will be hundreds of files in a station folder (multiple versions for each day of a pointing schedule), so the approach you are suggesting might get out of hand fairly quickly in terms of the size of the response to the `list_objects` query. However, we could mitigate that by archiving or deleting objects in the bucket 30 days after the date a file corresponds to has passed.
I think it's good to stick with the currently committed approach now, but if at some point we have performance issues, we can circle back to this and discuss the best way to improve.